yt_dlp/extractor/ertgr.py

```diff
@@ -0,0 +1,302 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import re
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    clean_html,
+    determine_ext,
+    ExtractorError,
+    dict_get,
+    int_or_none,
+    merge_dicts,
+    parse_qs,
+    parse_age_limit,
+    parse_iso8601,
+    str_or_none,
+    try_get,
+    unescapeHTML,
+    url_or_none,
+)
+
+
+class ERTFlixBaseIE(InfoExtractor):
+    _VALID_URL = r'ertflix:(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'ertflix:monogramma-praxitelis-tzanoylinos',
+        'md5': '5b9c2cd171f09126167e4082fc1dd0ef',
+        'info_dict': {
+            'id': 'monogramma-praxitelis-tzanoylinos',
+            'ext': 'mp4',
+            'title': 'md5:ef0b439902963d56c43ac83c3f41dd0e',
+        },
+    },
+    ]
```

Comment:
The extraction of actual ertflix videos is (still) delegated to it with the virtual URL scheme/handler `ertflix:<codename>`.

Reply:
but my point is that there is no reason to do that. The extractor can just directly call the relevent functions

Reply:
Yes, perhaps everything could be merged to a single extractor, but this allows you to directly extract if you know the `codename` (this may come in handy once they allow some form of embedding) and I think it doesn't hurt.

Reply:
ok, but inheriting an extractor from another user-facing extractor is generally bad idea. Either separate them, or make an actual BaseIE for both to inherit from

Reply:
OK, I will separate base extractor from this user-facing one then.

Reply:
I don't think the ertflix:codename pseodo-URL is really user-facing, more for testing and to expose the API to the command-line. In fact, shouldn't all XXXBaseIEs implement such a thing if it makes sense?

Reply:
> I think that has been replaced by an API call here but it could still be worth being able to get an item using its ertflix:ID from the command line or other code.

FWIW the added API call (GetTiles) does not replace the function of the base extractor.

Reply:
> more for testing and to expose the API to the command-line.

It is good to be able to test (bare) extraction if you only know `codename`.